hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/BalancerService.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.balancer;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HDFSBlocksDistribution;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.RegionInfo;
+import org.apache.hadoop.hbase.client.TableDescriptor;
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * This is the cluster we want to balance. It provides methods to let us get the information we
+ * want.
+ */
+@InterfaceAudience.Private
+public interface BalancerService {
+
+  List<RegionInfo> getAssignedRegions();
+
+  TableDescriptor getTableDescriptor(TableName tableName) throws IOException;
+
+  HDFSBlocksDistribution computeHDFSBlocksDistribution(Configuration conf,
+    TableDescriptor tableDescriptor, RegionInfo regionInfo) throws IOException;
+}
```

Comment:
After reading whole patch I see that this is not a BalancerService but a subset of the MasterSerivces, those methods the Balancer can't do without.

Suggest the class comment say that.

The getAssignedRegions is all assigned Regions on the cluster it seems? Add a bit of javadoc? Ditto for computeHDFSBlockDist...

Reply:
It is used to provide information which will be used for balancer. So I agree that BalancerService is not a good name. Maybe ClusterInfoRetrivier or ClusterInfoProvider?

Reply:
+1 on *Provider... ClusterInfoProvider is good.